### PR TITLE
Fixes acoustic range bearing orientation.

### PR DESCRIPTION
**Pull request by arjo129**
This PR fixes the range bearing transponder to follow the conventions sent to us by MBARI. In particular the range bearing system should calculate range-bearing in the following manner:
image

This PR introduces a new link that orients the acoustic transponder in the correct direction. Additionally it switches from traditional spherical coordinates to the one in the above diagram. 

We also have added a new test and example world with updated transponders. This test tests the correctness of the range-bearing program with respect to the model. The transponder placements are shown here:
transponder_labels

### CODE REVIEW

File: lrauv_ignition_plugins/src/RangeBearingPlugin.cc [hunk 1]
```diff
@@ -167,29 +169,45 @@ void RangeBearingPrivateData::PublishResponse(
...
+  auto negativeZAxis = ignition::math::Vector3d(0, 0, -1);
+  auto elev =
+    asin(negativeZAxis.Dot(poseInLocalFrame) / poseInLocalFrame.Length());
+
+  // Azimuth is given by the angle from the X axis in vehicle frame.
+  auto xyProj = ignition::math::Vector2d(poseInLocalFrame.X(),
+    poseInLocalFrame.Y());
+  // TODO(arjo): This minus sign shouldn't be necessary. 
+  auto azimuth = (xyProj.Length() < 0.001) ? 0 : -atan2(xyProj.Y(), xyProj.X());
+  igndbg << "theta " << elev << " phi " << azimuth << "\n";
```
Comment (braanan) [inline lrauv_ignition_plugins/src/RangeBearingPlugin.cc:202]: @arjo129 replace theta/phi with vehicle frame elevation/azimuth. For reference, the debug output in the LRAUV app looks like this:
```
2022-03-28T23:38:33.930Z,1648510713.930 [circle_acoustic_contact:B.](INFO): slant range: 51.063232 m
2022-03-28T23:38:33.930Z,1648510713.930 [circle_acoustic_contact:B.](INFO): rotation FSK->NED: [[-0.115604,0.992562,0.038176],[-0.952876,-0.121670,0.277892],[0.280470,-0.004252,0.959853]]
2022-03-28T23:38:33.930Z,1648510713.930 [circle_acoustic_contact:B.](INFO): direction FSK: [0.229616,0.897985,-0.375366]
2022-03-28T23:38:33.930Z,1648510713.930 [circle_acoustic_contact:B.](INFO): direction NED: [0.850431,-0.432364,-0.299714]
2022-03-28T23:38:33.930Z,1648510713.930 [circle_acoustic_contact:B.](INFO): rel pos FSK:   [11.724925,45.854018,-19.167385]
2022-03-28T23:38:33.930Z,1648510713.930 [circle_acoustic_contact:B.](INFO): rel pos NED:   [43.425751,-22.077918,-15.304351]
2022-03-28T23:38:33.930Z,1648510713.930 [circle_acoustic_contact:B.](INFO): FSK: azumith=75.656735 deg, elevation=-22.046916 deg.
2022-03-28T23:38:33.930Z,1648510713.930 [circle_acoustic_contact:B.](INFO): NED: bearing=-26.949063 deg, inclination=-17.440408 deg.
2022-03-28T23:38:33.930Z,1648510713.930 [circle_acoustic_contact:B.](INFO): NED: lat=36.803405 deg, lon=-121.827208 deg, depth=-0.423813 m.
```

File: lrauv_ignition_plugins/proto/lrauv_range_bearing_response.proto [hunk 2]
```diff
@@ -33,8 +33,8 @@ message LRAUVRangeBearingResponse
   /// \brief Request ID. Each request ID must be unique.
   uint32 req_id = 1;
 
-  /// \brief Bearing in terms of (r, theta, phi). This is based on ground truth
-  /// and not calculated using time-of-flight
+  /// \brief Bearing in terms of (r, elevation, azimuth). This is based
+  /// on ground truth and not calculated using time-of-flight
```
Comment (braanan) [inline lrauv_ignition_plugins/proto/lrauv_range_bearing_response.proto:37]: As we discussed, let's replace the bearing vector with range, azimuth, and elevation members with commentary describing the frame of ref and units.

Comment (braanan) [reply lrauv_ignition_plugins/proto/lrauv_range_bearing_response.proto:37]: Also, let's replace `double range = 3;` in 42 with the actual time-of-flight in seconds

Comment (braanan) [reply lrauv_ignition_plugins/proto/lrauv_range_bearing_response.proto:37]: These changes could (should?) be worked into a separate PR

**Review by braanan — CHANGES_REQUESTED** [review]

File: lrauv_ignition_plugins/proto/lrauv_range_bearing_response.proto (hunk 2 above)
Comment (arjo129) [reply lrauv_ignition_plugins/proto/lrauv_range_bearing_response.proto:37]: Yeah I've opened a new PR for these changes see #183

Comment (braanan) [reply lrauv_ignition_plugins/proto/lrauv_range_bearing_response.proto:37]: Thanks, @arjo129. Then let's fix the debug output in the other comment and should be good to go

**Review by braanan — APPROVED** [review]

**Pull request merged**